syntax = "proto3";

package RealTimeChat;

service Chat {
    rpc StartNew (ChatObject) returns (ChatReply);
    rpc Get (GetRequest) returns (GetReply);
    rpc StreamUpdates (stream StreamRequest) returns (stream StreamUpdate);
}

message ChatObject {
    string id = 1;
    string subject = 2;
    string date = 3;
    string user = 4;
    string message = 5;
}

message ChatReply {
    string id = 1;
    string message = 2;
    string date = 3;
}

message GetRequest {}

message GetReply {
    repeated ChatObject ret = 1;
}

message StreamRequest {
    string id = 1;
    bytes payload = 2;
}

message StreamUpdate {
    bytes payload = 1;
    int64 timestamp = 2;
    bool received = 3;
}